lib/rules/no-descending-specificity/index.js

```diff
@@ -92,23 +95,24 @@ const rule = (primary, secondaryOptions) => {
 		/**
 		 * @param {import('postcss-selector-parser').Root} selectorNode
 		 * @param {import('postcss').Rule} ruleNode
-		 * @param {Map<any, any>} comparisonContext
+		 * @param {Map<string, Entry[]>} comparisonContext
 		 */
 		function checkSelector(selectorNode, ruleNode, comparisonContext) {
 			const selector = selectorNode.toString();
-			const referenceSelectorNode = lastCompoundSelectorWithoutPseudoClasses(selectorNode);
+			const referenceSelector = lastCompoundSelectorWithoutPseudoClasses(selectorNode);
```

Comment:
[note] `lastCompoundSelectorWithoutPseudoClasses()` returns a string, not a node. So I think `referenceSelector` is a more readable name.